in_toto/verifylib.go

```diff
@@ -498,8 +498,23 @@ func VerifyLinkSignatureThesholds(layout Layout,
 			// If the signer's key wasn't in our step's pubkeys array, check the cert pool to
 			// see if the key is known to us.
 			if !isAuthorizedSignature {
+				sig, err := linkMb.GetSignatureForKeyId(signerKeyID)
+				if err != nil {
+					continue
+				}
+
+				cert, err := sig.GetCertificate()
+				if err != nil {
+					continue
+				}
+
+				// test certificate against the step's constraints to make sure it's a valid functionary
+				if !step.CheckCertConstraints(cert) {
+					continue
+				}
+
```

Comment:
Metablock.VerifyLinkSignatureThresholds really needs some TLC.  Punt to refactor :disappointed: